4 - Tables and Persistence/6 - Intermediate Table Views/lab/FavoriteBook/FavoriteBook/BookTableViewController.swift

```diff
@@ -35,19 +35,17 @@ class BookTableViewController: UITableViewController {
     }
     
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: PropertyKeys.bookCell, for: indexPath)
-        
+      let cell = tableView.dequeueReusableCell(withIdentifier: PropertyKeys.bookCell, for: indexPath) as! BookTableViewCell//need to downcast the cell in order to retrieve from BookTableViewCell to use update(with:)
         let book = books[indexPath.row]
-        cell.textLabel?.text = book.title
-        cell.detailTextLabel?.text = book.description
+        cell.update(with: book)
         
         return cell
     }
     
     // MARK: - Navigation
     
     @IBAction func prepareForUnwind(segue: UIStoryboardSegue) {
-        guard let source = segue.source as? BookFormViewController,
+        guard let source = segue.source as? BookFormTableViewController,
             let book = source.book else {return}
         
         if let indexPath = tableView.indexPathForSelectedRow {
```

Comment:
@Natasha-A  take a look over here, when you edit the book is it replacing the existing book as expected or creating a new item?

Reply:
fix edit flow